examples/seq2seq/utils.py

```diff
@@ -304,9 +306,15 @@ def __call__(self, batch) -> Dict[str, torch.Tensor]:
             labels = trim_batch(labels, self.pad_token_id)
             input_ids, attention_mask = trim_batch(input_ids, self.pad_token_id, attention_mask=attention_mask)
 
+        if isinstance(self.tokenizer, T5Tokenizer):
```

Comment:
Sorry, yeah this was my bad. I didn't get why we've inputted both `decoder_input_ids` and `labels` previously - should have thought about it a bit more!

It's a tough decision on how to solve this long-term I think:

- there is no guarantee that `shift_tokens_right` is the same for all models, we already have three different methods (MBart, EncoderDecoder (not yet implemented but will be different), Bart & other enc-dec)
- We don't really want to start an `if isinstance(self.tokenizer)`-else cascade in the data collactor.
- I don't really like to add a flag `_do_compute_loss` to the encoder-decoder models because it breaks consistency that only passing `lables` will trigger a loss calculation with all other models and adds more complexity to the user in that another config param has to be understood.
- Another option would be to pass the models to the `__init__` of the data collator and then to call: `decoder_input_ids = model.shift_right(...)` and forcing all functions to have the same name similar to `prepare_inputs_for_generation(...)` 
- I think overall, it would actually be better to get fully rid of the data collators all together and instead make use of `datasets.map(...)` to prepare the inputs and just use the default datacollator as shown in https://github.com/huggingface/transformers/blob/ca422e3d7d8086baf233993af2f6f90f30729933/src/transformers/data/data_collator.py#L35 => then we wouldn't have any of those problems and would have more control over what to input to the model

Maybe we should discuss this quickly offline

Reply:
> I didn't get why we've inputted both decoder_input_ids and labels previously

we prepared both `decoder_input_ids` and `labels` because we need to pass `decoder_input_ids` to model explicitly when doing label smoothing as we don't pass `labels` to avoid calculating the loss twice.